pravega/pravega
# Issue 3790: Cherry pick recent commits to r0.5

**Change log description**  
* Cherry-picks commits from `master` to `r0.5`.

**Purpose of the change**  
Fixes #3790 

**What the code does**  
Cherry-picks the following commits:

```
Issue 3735: Update StreamCut Validation inside a ReaderGroup to handle completed Segments. (#3740)
Issue 3775: (SegmentStore) Delete attributes from Tier 2 when lazily deleting segments (#3776)
Issue 3758: Fix typo in controller API call name (#3759)
Issue 3725: Fixed sporadic NullPointerException in MetricsTest. (#3780)
Issue 3781: VersionedSerializer support for Compacted Signed Longs (#3782)
Issue 3772: (SegmentStore) Fixed DurableLogTests.testTruncateWithoutRecovery (#3773)
Issue 3655: TOC and section on connection pooling included in Pravega-concepts.md (#3678)
Issue 3538: Fixing unit test StreamSegmentContainerTests.testWriteFenceOut (#3771)
Issue 3721: Fix Multicontroller test failure in docker swarm based system tests (#3722)
Issue 3768: Increase timeout for pod/task-pv-pod creation (#3769)
Issue 3715: replaced the non thread safe map with SimpleMeterRegistry (#3732)
Issue 3596:  Improved Exception handling for ReplyProcessor. (#3648)
Issue 3477: Fixes on EventProcessors during Controller graceful restart (#3481)
```

**How to verify it**  
It should build and contributors should validate that there is no inconsistent change with any of the commits picked.
